ReflectometryServer/beamline.py

```diff
@@ -192,9 +194,9 @@ def _validate(self, beamline_parameters, modes):
                 errors.append("Mode must be uniquely named. Duplicate '{}'".format(mode))
 
         for mode in modes:
-            errors.extend(mode.validate(self._beamline_parameters.keys()))
+            errors.extend(mode.validate(list(self._beamline_parameters.keys())))
```

Comment:
Explicit cast not required here. `x in dictview` is allowed with view data type.